commitment/tap.go

```diff
@@ -313,9 +313,6 @@ func IsTaprootAssetCommitmentScript(script []byte) bool {
 	if len(script) != TaprootAssetCommitmentScriptSize {
 		return false
 	}
-	if script[0] != byte(asset.V0) {
```

Comment:
Should this just check for V0 _or_ V1 instead? Or why is this removed?

Reply:
IIRC, was removed when we realized that this would actually fail for any v1 (segwit) commitments. 

I think the open question here is: do we need to enumerate all the current/known asset versions? Or is enough that we bind the greater structure (script size, and the asset market). 

My fear is that if we enumerate all the versions here, then we'll need to go back and update this each time for older clients, so not quite soft forkable. Instead, I think for the known versions, we can validate the requirements we know (in this case, the top level hash or w/e hasn't changed), then just return true otherwise. IMO we want to avoid the case where an old client can't swap or interact w/ a new client, as the new client wants their asset in a v3 commitment (uses sha3 or w/e).

Reply:
Moved to a separate commit with notes.